netlogo-core/src/main/agent/InRadiusOrCone.scala

```diff
@@ -0,0 +1,428 @@
+        if (dy > worldHeight2)
+          dy = worldHeight - dy
+
+        gRoot = world.rootsTable.gridRoot(dx * dx + dy * dy)
+
+        // Only check patches that might have turtles within the radius on them.
+        // The 1.415 (square root of 2) adjustment is necessary because it is
+        // possible for portions of a patch to be within the circle even though
+        // the center of the patch is outside the circle.  Both turtles, the
+        // turtle in the center and the turtle in the agentset, can be as much
+        // as half the square root of 2 away from its patch center.  If they're
+        // away from the patch centers in opposite directions, that makes a total
+        // of square root of 2 additional distance we need to take into account.
+        if (gRoot <= radius + 1.415) {
+          val turtleIterator = patch.turtlesHere.iterator
+          while (turtleIterator.hasNext) {
+            val turtle = turtleIterator.next()
+
+            var found = false
+            val offsetIterator = offsets.iterator
+            while (!found && offsetIterator.hasNext) {
+              val (offsetX, offsetY) = offsetIterator.next()
+              if ((sourceSetIsWorldTurtles || (sourceSetIsBreedSet && (sourceSet eq turtle.getBreed))
+                || cachedIDs.contains(turtle.id))
+                && isInCone(turtle.xcor + worldWidth * offsetX, turtle.ycor + worldHeight * offsetY, startTurtle.xcor, startTurtle.ycor, radius, half, startTurtle.heading)) {
+                result.add(turtle)
+                found = true
+              }
+            }
+          }
+        }
+        i += 1
+      }
+    }
+
+    result
+  }
+
+  // helper method for inCone()
+  // for each world W at offsets offsetX and offsetY, check if any point
+  // in W is within radius by checking W's closest point
+  private def closestPointIsInRadius(x: Double, y: Double, offsetX: Int, offsetY: Int, r: Double): Boolean = {
+    var closestX, closestY = .0
+
+    if (offsetY < 0) { // below current world
+      closestY = world.maxPycor + (offsetY * world.worldHeight)
+    } else if (offsetY > 0) { // above current world
+      closestY = world.minPycor + (offsetY * world.worldHeight)
+    } else { // same horizontal as current world
+      closestY = y
+    }
+
+    if (offsetX < 0) { // left of current world
+      closestX = world.maxPxcor + (offsetX * world.worldWidth)
+    } else if (offsetX > 0) { //right of current world
+      closestX = world.minPxcor + (offsetX * world.worldWidth)
+    } else { // same vertical as current world
+      closestX = x
+    }
+
+    world.protractor.distance(closestX, closestY, x, y, false) <= r
+  }
+
+  // helper method for inCone().
+  // check if (x, y) is in the cone with center (cx, cy) , radius r, half-angle half, and central
+  // line of the cone having heading h.
+  private def isInCone(x: Double, y: Double, cx: Double, cy: Double, r: Double, half: Double, h: Double): Boolean = {
```

Comment:
This method can be made `@scala.inline` here and on all `private` methods in this class.